apps/studio/components/interfaces/APIKeys/APIKeyRow.tsx

```diff
@@ -0,0 +1,193 @@
+import { PermissionAction } from '@supabase/shared-types/out/constants'
+import { useQueryClient } from '@tanstack/react-query'
+import { InputVariants } from '@ui/components/shadcn/ui/input'
+import { Eye, MoreVertical } from 'lucide-react'
+import { useState } from 'react'
+import { toast } from 'sonner'
+
+import { useParams } from 'common'
+import CopyButton from 'components/ui/CopyButton'
+import { useAPIKeyIdQuery } from 'data/api-keys/[id]/api-key-id-query'
+import { APIKeysData } from 'data/api-keys/api-keys-query'
+import { apiKeysKeys } from 'data/api-keys/keys'
+import { AnimatePresence, motion } from 'framer-motion'
+import { useCheckPermissions } from 'hooks/misc/useCheckPermissions'
+import {
+  Button,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuSeparator,
+  DropdownMenuTrigger,
+  TableCell,
+  TableRow,
+  cn,
+} from 'ui'
+import { APIKeyDeleteDialog } from './APIKeyDeleteDialog'
+
+export const APIKeyRow = ({
+  apiKey,
+}: {
+  apiKey: Extract<APIKeysData[number], { type: 'secret' | 'publishable' }>
+}) => {
+  const MotionTableRow = motion(TableRow)
+
+  return (
+    <MotionTableRow
+      layout
+      initial={{ opacity: 0, height: 0 }}
+      animate={{ opacity: 1, height: 'auto' }}
+      exit={{ opacity: 0, height: 0 }}
+      transition={{
+        type: 'spring',
+        stiffness: 500,
+        damping: 50,
+        mass: 1,
+      }}
+    >
+      <TableCell className="py-2">{apiKey.description || '/'}</TableCell>
+      <TableCell className="py-2">
+        <div className="flex flex-row gap-2">
+          <Input apiKey={apiKey} />
+        </div>
+      </TableCell>
+
+      <TableCell className="flex justify-end">
+        <DropdownMenu>
+          <DropdownMenuTrigger className="px-1 focus-visible:outline-none" asChild>
+            <Button
+              type="text"
+              size="tiny"
+              icon={
+                <MoreVertical size="14" className="text-foreground-light hover:text-foreground" />
+              }
+            />
+          </DropdownMenuTrigger>
+          <DropdownMenuContent className="max-w-40" align="end">
+            <APIKeyDeleteDialog apiKey={apiKey} />
+            <DropdownMenuSeparator />
+            <APIKeyDeleteDialog apiKey={apiKey} />
```

Comment:
Seeing 2 Dropdown options to delete API key here, for secret key
![image](https://github.com/user-attachments/assets/41809a7d-0eb0-47e5-a518-7c927baff482)

Reply:
fixed